### DISCUSSION TITLE
Using recursive workflow with different sets of inputs and outputs

### DISCUSSION THREAD

**Discussion by JoshuaEPSamuel** [nextflow-io/nextflow#5297] [Q&A]
Hi all,

I've been recently experimenting with the recursion feature to see if it's possible for nextflow to handle our cyclic workflow task.

The TL;DR of the linear workflow is:
- Each process is running inside their respective docker containers
- Generate data in first process
- Subsequent processes takes results from previous processes and generates more data
- Final process analyses all the data - keeps good results and discards bad results.

Now we want to upgrade this, such that the workflow has to generate X amount of good results. The idea is that each time the workflow is run, it will generate Y amount of good results which are stored in a specific directory; until the number of files in the specific directory reaches our requirement number X it will run the workflow over and over again, slowly disposing good results into the specific directory until complete.

Currently, I am stuck at the following error and am unsure if it possible to make this work for our task:
`Workflow 'recursive_workflow' inputs and outputs do not have the same cardinality - Feedback loop is not supported`

The inputs to the workflow are all of the parameters needed for each process, and there is a single output which is the shared directory depositing good results.

**Comment by bentsherman** [comment]
Because the process output channels are fed back directly into the input channels, the inputs and outputs have to match. The easiest way to make this work is to have one big tuple input that also includes any output files, even if they are just "dummy" files initially

**Comment by JoshuaEPSamuel** [comment]
Thanks for suggestion! Managed to make this work
